### DISCUSSION TITLE
persisting data

### DISCUSSION THREAD

**Discussion by sentriz** [flawiddsouza/Restfox#61] [Q&A] [answered]
hi! thanks for restfox its very nice

I'm wondering what the strategy is for persisting requests? say I've imported 500 requests from insomnia over 8 workspaces - it would be a bumber to lose them

for the web version, the data is lost when I clear my browser history.
for electron version, how does persistence work? I can't see any mention of filesystem paths

is there a strategy here, am I missing something?

thanks!

**Comment by flawiddsouza** [comment]
It uses IndexedDB for storing the data. Same for web and electron. You can backup your data by clicking on Workspaces:
![image](https://github.com/flawiddsouza/Restfox/assets/10164500/409247cf-0029-4773-9fc1-45a426f4266f)
![image](https://github.com/flawiddsouza/Restfox/assets/10164500/8be2458d-45dc-4c4e-852f-a92827d05d8e)

**Comment by sentriz** [comment]
ah i see. i was looking for a solution where the source of truth is the filesystem, so no manual export/import is needed. is there a issue for this, or would you consider this as an option? i can make an issue if so. thanks

**Reply by flawiddsouza** [reply]
Well, I'd have to implement a completely different way of storing the data for electron, which I currently don't have the bandwidth to work on. Feel free to open an issue. Maybe someone can take it up and send a PR for it.

**Reply by flawiddsouza** [reply, marked as answer]
Just to update, the desktop version now supports storing the data wherever you choose in the file system while creating a workspace.

**Reply by sentriz** [reply]
i noticed, i've been using it for weeks and works great :+1: thank you!
